beta/airconsole-1.10.0.js

```diff
@@ -611,13 +634,32 @@ AirConsole.prototype.convertDeviceIdToPlayerNumber = function(device_id) {
  */
 AirConsole.prototype.onDeviceMotion = function(data) {};
 
+
+
+/**
+ * @typedef {Object} CompositionVibrationData
+ * @property {string} primitive - Identifier used to play a specific vibration primitive
+ * @property {Number} scale - Vibration scale value between 0.0 and 1.0
+ * @property {Number} [delay] - Delay in milliseconds before this primitive is played
+ */
+
+/**
+ * @typedef {Object} VibrationOptions
+ * @property {string} type - Type of abstraction interface to use for executing
+ * vibrations. This should currently always be set to 'composition'.
+ * @property {CompositionVibrationData[]} value - Array of vibration parameters depending
+ * on the interface type
+ */
+
 /**
- * Vibrates the device for a specific amount of time. Only works for controllers.
- * Note: iOS ignores the specified time and vibrates for a pre-set amount of time.
- * @param {Number} time - Milliseconds to vibrate the device
+ * Vibrates the device for a specific amount of time or playing back a specific pattern when
+ * used with the controller app. Only works for controllers.
+ * Note: iOS without controller app ignores the specified time and vibrates for a pre-set amount of time.
+ * @param {Number} options - Milliseconds to vibrate the device
```

Comment:
Due to how different tools interpret this (even making up additional options parameters), I would recommend to use the type union approach for the documentation.

```suggestion
 * @param {Number|VibrationOptions} options: This represents either <br />
                                                                                 - Milliseconds to vibrate the device <br />
                                                                                 - Vibration options for fine-tuned vibration patterns.
```

you would probably need to format this locally due to the suggestion interface formatting